src/backend/api/meals.js

```diff
@@ -5,8 +5,144 @@ const knex = require("../database");
 router.get("/", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    // const titles = await knex("meals").select("title");
+
+    //maxPrice
+    if (request.query.maxPrice) {
+      const queryMaxPrice = parseInt(request.query.maxPrice);
+      const mealsWithPrice = await knex("meal").where(
+        "price",
+        "<",
+        queryMaxPrice
+      );
+      response.json(mealsWithPrice);
+
+      if (isNaN(queryMaxPrice)) {
+        return response.status(400).send("Enter a valid id");
+      }
+    }
+
+    //availableReservations
+    else if (request.query.availableReservations === "true") {
+      let coalesceres = knex.raw(
+        "coalesce(sum(reservation.number_of_guests), 0) as total_reservation"
+      );
+      const availableReservations = await knex("meal")
+      //  let coalesceres =
+        .select("meal.id", "max_reservations", coalesceres)
+        .leftJoin("reservation", "reservation.meal_id", "meal.id")
+        .groupBy("meal.id")
+        .having("max_reservations", ">", "total_reservation");
+      response.json(availableReservations);
+    }
+
+    //title
+    else if (request.query.title) {
+      const queryTitle = request.query.title;
+
+      const mealWithtitleSearch = await knex("meal").where(
+        "title",
+        "like",
+        `%${queryTitle}%`
+      );
+
+      response.json(mealWithtitleSearch);
+    }
+
+    //createdAfter
+    else if (request.query.createdAfter) {
+      const querycreatedAfter = new Date(request.query.createdAfter);
+
+      if (isNaN(querycreatedAfter)) {
+        return response.status(400).send("Enter a valid date");
+      }
```

Comment:
This should also be above
```
const querycreatedAfter = new Date(request.query.createdAfter);
```
Otherwise, an error will already have occurred when trying to convert the createdAfter to Date and this if statement will not be reached!